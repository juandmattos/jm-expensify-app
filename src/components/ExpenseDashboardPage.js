import React from 'react'
import ReactDOM from 'react-dom'
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'
import ExpensesSummary from './ExpensesSummary'

const ExpenseDashboardPage = () => (
  <div>
    <ExpensesSummary />
    <ExpenseList />
    <ExpenseListFilters />
  </div>
)

export default ExpenseDashboardPage
