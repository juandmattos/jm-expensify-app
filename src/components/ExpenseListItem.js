import React from 'react'
import { Link } from 'react-router-dom'

export const ExpenseListItem = ({ id, description, amount, createdAt, removeExpense, history }) => (
    <div>
      <Link to={`/edit/${id}`}>
        <h3>{ `Expense: ${description}` }</h3>
      </Link>
      <p>{ `Amount: ${amount} - CreatedAt: ${createdAt}` }</p>
    </div>
)

export default ExpenseListItem
