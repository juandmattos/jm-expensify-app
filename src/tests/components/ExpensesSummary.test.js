import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'
import expenses from '../fixtures/expenses'
import selectExpensesTotal from '../../selectors/expenses-total'

test('should render ExpensesSummary', () => {
  const wrapper = shallow(
    <ExpensesSummary
      expenseCount={expenses.length}
      expensesTotal={selectExpensesTotal(expenses)}
    />
  )

  expect(wrapper).toMatchSnapshot()
})