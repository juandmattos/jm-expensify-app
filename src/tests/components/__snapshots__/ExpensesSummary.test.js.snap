// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary 1`] = `
<div>
  <h1>
    Viewing 3 expenses
        totalling $1,141.95
  </h1>
</div>
`;
